app/pages/sections/call-off-ordering-party/controller.js

```diff
@@ -1,8 +1,32 @@
-import { getData } from 'buying-catalogue-library';
+import { getData, putData } from 'buying-catalogue-library';
 import { getContext } from './contextCreator';
 import { getEndpoint } from '../../../endpoints';
 import { logger } from '../../../logger';
 
+const formatPutData = data => ({
+  organisation: {
+    name: data.name ? data.name.trim() : null,
+    odsCode: data.odsCode ? data.odsCode.trim() : null,
+    address: {
+      line1: data.line1 ? data.line1.trim() : null,
+      line2: data.line2 ? data.line2.trim() : null,
+      line3: data.line3 ? data.line3.trim() : null,
+      line4: data.line4 ? data.line4.trim() : null,
+      line5: data.line5 ? data.line5.trim() : null,
+      town: data.town ? data.town.trim() : null,
+      county: data.county ? data.county.trim() : null,
+      postcode: data.postcode ? data.postcode.trim() : null,
+      country: data.country ? data.country.trim() : null,
+    },
+  },
+  primaryContact: {
+    firstName: data.firstName ? data.firstName.trim() : null,
+    lastName: data.lastName ? data.lastName.trim() : null,
+    emailAddress: data.emailAddress ? data.emailAddress.trim() : null,
+    telephoneNumber: data.telephoneNumber ? data.telephoneNumber.trim() : null,
+  },
+});
```

Comment:
In the template for this page. If you set the id of these fields and prefix it to include its parent id eg `address` or `primaryContact` then this should give you this structure when you come to submit it. Rather than a flat structure that you are getting at the moment and then having to do this format.

Reply:
ooh interesting

Reply:
On second thoughts I'm not sure it's going to be actually worth it. I'd leave as is for now. It's better I think that we have control on the structure of the body as you have done. Rather than rely on the form.

Reply:
ok that is easier for me 😄